packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/aws.ts

```diff
@@ -0,0 +1,167 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { IStateMachine } from '@aws-cdk/aws-stepfunctions';
+import { Construct } from 'constructs';
+import { AwsServiceIntegration, AwsServiceIntegrationProps } from './private/integration';
+
+/**
+ * The common Step Functions integration resource for HTTP API
+ */
+abstract class StepFunctionsIntegration extends AwsServiceIntegration {
+
+  /**
+   *
+   * @internal
+   */
+  protected _integrationService(): string {
+    return 'StepFunctions';
+  }
+}
+
+/**
+ * Step Functions StartExecution integration properties
+ */
+export interface StepFunctionsStartExecutionIntegrationProps extends AwsServiceIntegrationProps {
+
+  /**
+   * The state machine to be executed
+   */
+  readonly stateMachine: IStateMachine;
+
+  /**
+   * The execution name
+   *
+   * @default - undefined
+   */
+  readonly name?: string;
+
+  /**
+   * The input parameters of execution
+   *
+   * @default - undefined
+   */
+  readonly input?: any;
+
+  /**
+   * The region of state machine
+   *
+   * @default - undefined
+   */
+  readonly region?: string;
+}
+
+/**
+ * The StepFunctions-StartExecution integration resource for HTTP API
+ */
+export class StepFunctionsStartExecutionIntegration extends StepFunctionsIntegration {
```

Comment:
this one says StartExecution but later on L79 you are giving permission to StartSyncExecution.

Also resources [*] should be avoided if you have the state machine.

Reply:
Good Note. I will fix this. Thank you